src/GameController.cpp

```diff
@@ -212,9 +213,73 @@ void GameController::handlePlayerMovement(const short& movement)
         return;
     }
 
+    /* check if the player is against a wall during the movement; if there is
+       a collision, the wall is displayed by the condition function and the
+       movement is cancelled */
+    if (level.detectWalls(movement))
```

Comment:
on the algorithm hand, the two previous conditions could be refactored into one; explain why they are still separated (order)